### ISSUE TITLE
Error generating the Frequency/Recency Matrix

### ISSUE THREAD

**Issue by joaolcorreia** [CamDavidsonPilon/lifetimes#36]
Hi,

When trying to generate the Frequency/Recency Matrix I get an error.
I'm using Anaconda matplotlib-1.5. Any idea how to fix this?

```
AttributeError                            Traceback (most recent call last)
<ipython-input-4-7b0ad3a728a5> in <module>()
      1 from lifetimes.plotting import plot_frequency_recency_matrix
      2 
----> 3 plot_frequency_recency_matrix(bgf)

//anaconda/lib/python2.7/site-packages/lifetimes/plotting.pyc in plot_frequency_recency_matrix(model, T, max_frequency, max_recency, **kwargs)
    116     # necessary for colorbar to show up
    117     PCM = ax.get_children()[2]
--> 118     plt.colorbar(PCM, ax=ax)
    119 
    120     return ax

//anaconda/lib/python2.7/site-packages/matplotlib/pyplot.py in colorbar(mappable, cax, ax, **kw)
   2235         ax = gca()
   2236 
-> 2237     ret = gcf().colorbar(mappable, cax = cax, ax=ax, **kw)
   2238     return ret
   2239 colorbar.__doc__ = matplotlib.colorbar.colorbar_doc

//anaconda/lib/python2.7/site-packages/matplotlib/figure.py in colorbar(self, mappable, cax, ax, use_gridspec, **kw)
   1593                 cax, kw = cbar.make_axes(ax, **kw)
   1594         cax.hold(True)
-> 1595         cb = cbar.colorbar_factory(cax, mappable, **kw)
   1596 
   1597         self.sca(current_ax)

//anaconda/lib/python2.7/site-packages/matplotlib/colorbar.py in colorbar_factory(cax, mappable, **kwargs)
   1328         cb = ColorbarPatch(cax, mappable, **kwargs)
   1329     else:
-> 1330         cb = Colorbar(cax, mappable, **kwargs)
   1331 
   1332     cid = mappable.callbacksSM.connect('changed', cb.on_mappable_changed)

//anaconda/lib/python2.7/site-packages/matplotlib/colorbar.py in __init__(self, ax, mappable, **kw)
    878         # Ensure the given mappable's norm has appropriate vmin and vmax set
    879         # even if mappable.draw has not yet been called.
--> 880         mappable.autoscale_None()
    881 
    882         self.mappable = mappable

AttributeError: 'Spine' object has no attribute 'autoscale_None'
```

**Comment by joaolcorreia** [comment]
On /lifetimes/plotting.py commenting the bar prints the graph but with no scale.

#PCM = ax.get_children()[2]
#plt.colorbar(PCM, ax=ax)

**Comment by CamDavidsonPilon** [comment]
thanks for reporting this, I'll take a look =)

**Comment by rhydomako** [comment]
I've also run into this problem with Matplotlib 1.5.0

I found that if I hold onto the imshow object

```
p = ax.imshow(Z, interpolation=interpolation, **kwargs)
```
and then pass that object to the colorbar function:
```
plt.colorbar(p)
```
then it will correct output both the plot and colorbar without having to call _get_children_.

I also had to make this change for the _plot_probability_alive_matrix_ function.

**Comment by joaolcorreia** [comment]
I installed matplotlib-1.4.3 and the problem was solved. The issue is in matplotlib-1.5.0 and above.

**Comment by CamDavidsonPilon** [comment]
Hmm, I'm not able to reproduce this locally. (I upgrade to 1.5.1 just now).

**Comment by rhydomako** [comment]
This should be fixed by #37 (I just forgot to mention it here).

**Comment by CamDavidsonPilon** [comment]
whoops, I missed that - thanks @rhydomako!
